receiver/awss3receiver/receiver.go

```diff
@@ -4,24 +4,84 @@
 package awss3receiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awss3receiver"
 
 import (
+	"bytes"
+	"compress/gzip"
 	"context"
+	"io"
+	"strings"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
 	"go.uber.org/zap"
 )
 
-type awss3Receiver struct {
+type awss3TraceReceiver struct {
+	s3Reader *s3Reader
+	consumer consumer.Traces
+	logger   *zap.Logger
+	cancel   context.CancelFunc
 }
 
-func newAWSS3TraceReceiver(_ *Config, _ consumer.Traces, _ *zap.Logger) (*awss3Receiver, error) {
-	return &awss3Receiver{}, nil
+func newAWSS3TraceReceiver(ctx context.Context, cfg *Config, traces consumer.Traces, logger *zap.Logger) (*awss3TraceReceiver, error) {
+	reader, err := newS3Reader(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &awss3TraceReceiver{
+		s3Reader: reader,
+		consumer: traces,
+		logger:   logger,
+		cancel:   nil,
+	}, nil
 }
 
-func (r *awss3Receiver) Start(_ context.Context, _ component.Host) error {
+func (r *awss3TraceReceiver) Start(ctx context.Context, _ component.Host) error {
```

Comment:
```suggestion
func (r *awss3TraceReceiver) Start(_ context.Context, _ component.Host) error {
```
I believe lint may fail on this line if `ctx` isn't used.

Reply:
It won't but for the not very nice reason that I've reused the `ctx` variable on the next line. I'll push another small update to make it clearer that the context passed in is not used.

Reply:
Thank you!